engine/src/module-engine.cpp

```diff
@@ -559,34 +559,83 @@ extern "C" MC_DLLEXPORT_DEF void MCEngineEvalMessageWasNotHandled(bool& r_not_ha
 
 ////////////////////////////////////////////////////////////////////////////////
 
-extern "C" MC_DLLEXPORT_DEF MCValueRef MCEngineExecExecuteScript(MCStringRef p_script)
+static MCValueRef
+MCEngineDoExecuteScriptInObjectWithArguments(MCStringRef p_script, MCObject *p_object, MCProperListRef p_arguments)
 {
-    if (!MCEngineScriptObjectAccessIsAllowed())
-        return nil;
-    
-    MCStack *t_stack;
-    t_stack = MCdefaultstackptr;
-    if (t_stack == nil)
-    {
-        MCErrorCreateAndThrow(kMCGenericErrorTypeInfo, "reason", MCSTR("no default stack"), nil);
-        return nil;
-    }
-    
-    MCRedrawLockScreen();
-    
-    Exec_stat t_stat;
-    t_stat = t_stack -> getcurcard() -> domess(p_script);
-    
-    MCRedrawUnlockScreen();
-    
-    if (t_stat == ES_ERROR)
-    {
-        MCEngineThrowScriptError();
-        return nil;
-    }
+	if (p_object == nil)
+	{
+		MCStack *t_stack = MCdefaultstackptr;
+		if (t_stack == nil)
+		{
+			MCErrorCreateAndThrow(kMCGenericErrorTypeInfo, "reason", MCSTR("no default stack"), nil);
+			return nil;
+		}
+		
+		p_object = t_stack->getcurcard();
+	}
```

Comment:
```c++
MCStack *t_stack = MCdefaultstackptr;
```
is equivalent to
```c++
MCStack *t_stack = MCdefaultstackptr.Get();
```
see [implementation](https://github.com/livecode/livecode/blob/develop/engine/src/object.h#L362).  This can assert.  To avoid a possible assertion failure, I think what you're looking for here is something like:
```c++
if (p_object == nil)
{
    if (MCdefaultstackptr) {
        p_object = MCdefaultstackptr->getcurcard();
    } else {
        MCErrorCreateAndThrow(kMCGenericErrorTypeInfo,
                              "reason", MCSTR("no default stack"), nil);
        return nullptr;
    }
}
```